packages/grid/x-grid/src/tests/reorder.XGrid.test.tsx

```diff
@@ -206,10 +215,151 @@ describe('<XGrid /> - Reorder', () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+      const dragCol = getColumnHeaderCell(1).firstChild!;
+      const targetCol = getColumnHeaderCell(0).firstChild!;
+
+      fireEvent.dragStart(dragCol);
+      fireEvent.dragEnter(targetCol);
+      const dragOverEvent = createDragOverEvent(targetCol);
+      fireEvent(targetCol, dragOverEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+
+      const dragEndEvent = createDragEndEvent(dragCol);
+      fireEvent(dragCol, dragEndEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+    });
+
+    it('should not allow to drag left of first visible column if it has disableReorder=true', () => {
+      let apiRef: GridApiRef;
+      const rows = [{ id: 0, brand: 'Nike' }];
+      const columns = [
+        { field: 'brand', disableReorder: true },
+        { field: 'desc' },
+        { field: 'type' },
+      ];
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+      const dragCol = getColumnHeaderCell(1).firstChild!;
+      const targetCol = getColumnHeaderCell(0).firstChild!;
+
+      fireEvent.dragStart(dragCol);
+      fireEvent.dragEnter(targetCol);
+      const dragOverEvent = createDragOverEvent(targetCol);
+      fireEvent(targetCol, dragOverEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+
+      const dragEndEvent = createDragEndEvent(dragCol);
+      fireEvent(dragCol, dragEndEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+    });
+
+    it('should not allow to drag right of last visible column if it has disableReorder=true', () => {
+      let apiRef: GridApiRef;
+      const rows = [{ id: 0, brand: 'Nike' }];
+      const columns = [
+        { field: 'brand' },
+        { field: 'desc' },
+        { field: 'type', disableReorder: true },
+      ];
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+      const dragCol = getColumnHeaderCell(1).firstChild!;
+      const targetCol = getColumnHeaderCell(0).firstChild!;
```

Comment:
The index of the last visible column is 2, no?

```suggestion
      const targetCol = getColumnHeaderCell(2).firstChild!;
```